src/main/java/streamingapi/client/StreamMonitor.java

```diff
@@ -0,0 +1,28 @@
+package streamingapi.client;
+
+/**
+ * Monitor used stop the execution or to check if the execution should continue.
+ *
+ * @author dancojocaru
+ */
+public class StreamMonitor {
+
+	private boolean isActive = true;
+
+	/**
+	 * Stops the monitored execution
+	 * {@code false} represents
+	 */
+	public void stopExecution() {
```

Comment:
This is never used. Is this intended ?

Reply:
Right now, yes. I added it just as an example. To use it, a new async execution of the StreamingApiClient should be started, and then the main thread could stop it.